### PR TITLE
Fix right-to-left language support for voice recording

### PR DESCRIPTION
**Pull request by veeti**
### Contributor checklist

- [X] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [X] I have tested my contribution on these devices:
 * Nexus 5X, 7.1.1
- [X] My contribution is fully baked and ready to be merged as is
- [X] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)
- [X] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

----------

### Description

[Demo](https://my.mixtape.moe/wfrxpv.mp4)

### CODE REVIEW

File: src/org/thoughtcrime/securesms/components/InputPanel.java [hunk 1]
```diff
@@ -260,6 +264,11 @@ public void moveTo(float x) {
       slideToCancelView.startAnimation(animation);
     }
 
+    float getOffset(float x) {
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/components/InputPanel.java]: can this be private?

File: src/org/thoughtcrime/securesms/components/InputPanel.java [hunk 2]
```diff
@@ -260,6 +264,11 @@ public void moveTo(float x) {
...
+      return ViewCompat.getLayoutDirection(slideToCancelView) == LAYOUT_DIRECTION_LTR ?
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/components/InputPanel.java]: Shouldn't this be ViewCompat.LAYOUT_DIRECTION_LTR?

File: src/org/thoughtcrime/securesms/components/MicrophoneRecorderView.java [hunk 3]
```diff
@@ -179,6 +184,16 @@ public void hide(float x) {
...
 
+    float getOffset(float x) {
+      return ViewCompat.getLayoutDirection(recordButtonFab) == ViewCompat.LAYOUT_DIRECTION_LTR ?
+          -Math.max(0, this.startPositionX - x) : Math.max(0, x - this.startPositionX);
+    }
+
+    int getWidthAdjustment() {
+      int width = recordButtonFab.getWidth() / 4;
+      return ViewCompat.getLayoutDirection(recordButtonFab) == LAYOUT_DIRECTION_LTR ? -width : width;
+    }
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/components/MicrophoneRecorderView.java]: can these be private?

File: src/org/thoughtcrime/securesms/components/MicrophoneRecorderView.java [hunk 4]
```diff
@@ -179,6 +184,16 @@ public void hide(float x) {
...
     }
 
+    float getOffset(float x) {
+      return ViewCompat.getLayoutDirection(recordButtonFab) == ViewCompat.LAYOUT_DIRECTION_LTR ?
+          -Math.max(0, this.startPositionX - x) : Math.max(0, x - this.startPositionX);
+    }
+
+    int getWidthAdjustment() {
+      int width = recordButtonFab.getWidth() / 4;
+      return ViewCompat.getLayoutDirection(recordButtonFab) == LAYOUT_DIRECTION_LTR ? -width : width;
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/components/MicrophoneRecorderView.java]: Shouldn't this be ViewCompat.LAYOUT_DIRECTION_LTR?

**Comment by veeti** [comment]
Fixed! I also put in another tweak for a really minor "glitch" where the [elapsed time starts in Western numerals](https://gfycat.com/HollowDopeyInsect).

**Comment by moxie0** [comment]
hey, they are called arabic numerals after all =)

**Pull request closed without merging**